ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.ledger.api.auth
+
+import java.time.Clock
+
+import com.digitalasset.ledger.api.auth.interceptor.AuthorizationInterceptor
+import com.digitalasset.ledger.api.v1.transaction_filter.TransactionFilter
+import com.digitalasset.platform.server.api.validation.ErrorFactories.permissionDenied
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.Future
+
+object Authorizer {
+
+  private def authError = permissionDenied("You are not authorized to use this resource")
+
+}
+
+/** A simple helper that allows services to use authorization claims
+  * that have been stored by [[AuthorizationInterceptor]].
+  */
+final class Authorizer(clock: Clock) {
+
+  import Authorizer.authError
+
+  def requirePublicClaimsOnStream[Req, Res](
+      call: (Req, StreamObserver[Res]) => Unit): (Req, StreamObserver[Res]) => Unit =
+    wrapStream(_.isPublic(clock), call)
+
+  def requirePublicClaims[Req, Res](call: Req => Future[Res]): Req => Future[Res] =
```

Comment:
These methods are easier to use than before, but we might need to refactor them once more when we add the feature that the Authorizer fills in missing request information.

Reply:
I'll be more than happy to find a way to make this work. Hopefully the component that checks claims and the one that fills in information from the claims to the request can be pulled apart.